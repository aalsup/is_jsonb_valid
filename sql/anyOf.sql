-- anyOf
-- first anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '1');
-- second anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '2.5');
-- both anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '3');
-- neither anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '1.5');
-- anyOf with base schema
-- mismatch base schema
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '3');
-- one anyOf valid
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '"foobar"');
-- both anyOf invalid
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '"foo"');