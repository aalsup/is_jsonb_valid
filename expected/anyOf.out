-- anyOf
-- first anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '1');
 is_jsonb_valid 
----------------
 f
(1 row)

-- second anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '2.5');
 is_jsonb_valid 
----------------
 f
(1 row)

-- both anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '3');
 is_jsonb_valid 
----------------
 f
(1 row)

-- neither anyOf valid
SELECT is_jsonb_valid('{"anyOf":[{"type":"integer"},{"minimum":2}]}', '1.5');
 is_jsonb_valid 
----------------
 f
(1 row)

-- anyOf with base schema
-- mismatch base schema
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '3');
 is_jsonb_valid 
----------------
 f
(1 row)

-- one anyOf valid
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '"foobar"');
 is_jsonb_valid 
----------------
 f
(1 row)

-- both anyOf invalid
SELECT is_jsonb_valid('{"type":"string","anyOf":[{"maxLength":2},{"minLength":4}]}', '"foo"');
 is_jsonb_valid 
----------------
 f
(1 row)
